framework/src/main/java/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -1010,15 +1016,36 @@ private void checkContractsAtMethodDeclaration(
             try {
                 exprJe = JavaExpressionParseUtil.parse(expressionString, jeContext);
             } catch (JavaExpressionParseException e) {
-                exprJe = null;
-                checker.report(methodTree, e.getDiagMessage());
+
+                Matcher m =
+                        FORMAL_PARAM_NAME_WARNING_PATTERN.matcher(
+                                e.getDiagMessage().getArgs()[0].toString());
```

Comment:
This assumes the array is never empty (there is always a 0th element).  That is true currently but might not be in the future.  Using a more specific error message key would solve this, or you could check the array size before doing this.